pymeasure/instruments/tektronix/tek371A.py

```diff
@@ -0,0 +1,859 @@
+# copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in
+# all copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NON INFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+# THE SOFTWARE.
+#
+import logging
+import time
+from pyvisa import constants as pyvisa_constants
+
+from pymeasure.instruments import Instrument
+from pymeasure.instruments.validators import strict_range
+from pymeasure.instruments.validators import strict_discrete_set
+
+log = logging.getLogger(__name__)
+log.addHandler(logging.NullHandler())
+
+
+class WaveformPreamble:
+    """ Represents the waveform preamble for a curve data from the Tektronix model 371A.
+    The preamble contains the information needed for interpreting, scaling,
+    and labeling the numeric information of the curve.
+    This preamble is coded in ASCII characters and is readable by the operator
+    without interpretation by the controller.
+    """
+
+    # The preamble and curve are each a string of eight-bit bytes.
+    # The preamble is a string of ASCII letters, numerals, and punctuation.
+    # Each character is represented by one byte.
+
+    # Preambles are necessary to interpret the numeric information in the curve data that follows
+    # them. Within a preamble, 26 parameters are specified. The first ten are unique to the 371B
+    # curve tracer and are included as a sub-string linked to the WFID: label. The other 16
+    # parameters include ten that have fixed values and six that vary with the particular data
+    # sent. Within the WFID: sub-string the parameters are separated by slashes, while the entire
+    # sub-string is delimited by a pair of double quote marks. Most of the WFID: string is rather
+    # strictly defined, with each parameter value being right justified in a fixed length field.
+    # An exception is the BGM value, which may vary in field length. The remainder of the
+    # preamble uses standard punctuation. A colon links each parameter label with its
+    # corresponding value and the individual label and value pairs are separated with commas.
+
+    # A complete preamble might look like this: WFMPRE WFID:”INDEX 3/VERT 500MA/ HORIZ 1V/STEP
+    # 5V/OFFSET 0.00V/BGM 100mS/VCS 12.3/TEXT /HSNS VCE”, ENCDG:BIN, NR.PT:3,PT.FMT:XY,
+    # XMULT:+1.0E–2,XZERO:0,XOFF: 12,XUNIT:V,YMULT:+5.0E–3,YZERO:0,YOFF:12, YUNIT:A,BYT/ NR:2,
+    # BN.FMT:RP,BIT/NR:10,CRVCHK:CHKSMO,LN.FMT:DOT
```

Comment:
why don't you include this in the device docstring as well?
That way the API contains also this relevant information.

Reply:
got it. Thanks